src/components/scenes/SendScene2.tsx

```diff
@@ -189,6 +189,9 @@ const SendComponent = (props: Props) => {
   const parentExchangeDenom = getExchangeDenom(coreWallet.currencyConfig, null)
   const iconColor = useIconColor({ pluginId, tokenId })
 
+  const insufficientFeesModalPromise = React.useRef<(() => Promise<void>) | undefined>(undefined)
```

Comment:
This needs another name, because it's not a promise, but rather a function. Maybe call it `pendingInsufficientFeesModal`, or even change the type to `React.useRef<InsufficientFundsError | undefined>` and then copy / past the actual modal display logic to the two places. That way it's just plain-old data, not hidden logic.